### PR TITLE
feat(app):  add wifi setting to robot settings

### PR DESCRIPTION
**Pull request by koji**
# Overview
This PR is to make wifi setting components for unboxing flow available from RobotSettings and remove ODD menu from RobotSettings. I think we still need to keep ODD menu component for debug.

`WifiConnectionDetails` component that make available wifi settings from RobotSettings is messy right now. I will refactor this component by a following PR later.

# Test Plan
- Tap `Settings`
- Tap `NetworkSettings`
- Tap `Wi-Fi`
Try to set up wifi settings again

# Changelog
- update WifiConnectionDetails
- update SetWifiCred header part  to align with the hi-fi RobotSettings
- update SelectAuthenticationType header part to align with the hi-fi RobotSettings
- remove ODD menu link from RobotSettings

# Review requests

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [ ] merge https://github.com/Opentrons/opentrons/pull/12669

**Comment by koji** [comment]
@shlokamin 


https://github.com/Opentrons/opentrons/assets/474225/99a2e9c5-dbce-4f0d-a1d8-5057a287bd01

File: app/src/organisms/NetworkSettings/DisplaySearchNetwork.tsx [hunk 1]
```diff
@@ -8,28 +8,30 @@ import {
...
       <Flex
         justifyContent={JUSTIFY_CENTER}
         alignItems={ALIGN_CENTER}
         flexDirection={DIRECTION_COLUMN}
       >
         <StyledText
-          fontSize="2rem"
+          as="h3"
           fontWeight={TYPOGRAPHY.fontWeightSemiBold}
           lineHeight="2.72375rem"
```
Comment (jerader) [inline app/src/organisms/NetworkSettings/DisplaySearchNetwork.tsx]: is this line height needed?

File: app/src/organisms/NetworkSettings/DisplayWifiList.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+              borderRadius={BORDERS.size3}
+              css={NETWORK_ROW_STYLE}
+              flexDirection={DIRECTION_ROW}
+              padding={`${SPACING.spacing20} ${SPACING.spacing32}`}
+              alignItems={ALIGN_CENTER}
+              gridGap={SPACING.spacing4}
+              onClick={() => handleClick(nw)}
+            >
+              <Icon name="wifi" size="2.5rem" />
+              <StyledText as="h4" fontWeight={TYPOGRAPHY.fontWeightRegular}>
```
Comment (jerader) [inline app/src/organisms/NetworkSettings/DisplayWifiList.tsx]:
```suggestion
              <StyledText as="h4">
```

File: app/src/organisms/NetworkSettings/DisplayWifiList.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        borderRadius={BORDERS.size4}
+        color={COLORS.black}
+        css={NETWORK_ROW_STYLE}
+        padding={`${SPACING.spacing20} ${SPACING.spacing32}`}
+        flexDirection={DIRECTION_ROW}
+        alignItems={ALIGN_CENTER}
+        gridGap={SPACING.spacing4}
+      >
+        <Icon name="plus" size="2.5rem" color={COLORS.darkBlack100} />
+        <StyledText as="h4" fontWeight={TYPOGRAPHY.fontWeightRegular}>
```
Comment (jerader) [inline app/src/organisms/NetworkSettings/DisplayWifiList.tsx]:
```suggestion
        <StyledText as="h4" >
```

File: app/src/organisms/NetworkSettings/SelectAuthenticationType.tsx [hunk 4]
```diff
@@ -57,11 +62,16 @@ export function SelectAuthenticationType({
     showAlternativeSecurityTypeModal,
     setShowAlternativeSecurityTypeModal,
   ] = React.useState<boolean>(false)
+  const isInitialSetup = useIsFinishedUnboxing()
+
+  console.log('isInitialSetup', isInitialSetup)
```
Comment (jerader) [inline app/src/organisms/NetworkSettings/SelectAuthenticationType.tsx]: maybe we don't need this console log?

File: app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx [hunk 5]
```diff
@@ -47,6 +71,135 @@ export function WifiConnectionDetails({
...
+    const options = {
+      ssid: selectedSsid,
+      securityType: selectedAuthType,
+      hidden:
+        selectedAuthType === 'none' ||
+        (changeState.type === JOIN_OTHER && false),
+      psk: password,
+    }
+    dispatchApiRequest(Networking.postWifiConfigure(robotName, options))
+    // Note: kj 1/18/2023 for join_other network , this will be required by a following PR
```
Comment (jerader) [inline app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx]: can this note be deleted?

File: app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx [hunk 6]
```diff
@@ -47,6 +71,135 @@ export function WifiConnectionDetails({
   const subnetMask = wifi?.subnetMask != null ? wifi.subnetMask : noData
   const macAddress = wifi?.macAddress != null ? wifi.macAddress : noData
 
+  // ToDo (kj:05/24/2023) need to refactor this component
```
Comment (jerader) [inline app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx:74]: i agree, this component is sort of hard to follow!

File: app/src/pages/OnDeviceDisplay/ConnectViaWifi.tsx [hunk 7]
```diff
@@ -131,6 +132,7 @@ export function ConnectViaWifi(): JSX.Element {
         <WifiConnectionDetails
           ssid={changeState.ssid}
           authType={selectedAuthType}
+          // setShowInterfaceTitle={setShowInterfaceTitle}
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/ConnectViaWifi.tsx]: is this line not needed anymore?

File: app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx (hunk 6 above)
Comment (koji) [reply app/src/organisms/OnDeviceDisplay/RobotSettingsDashboard/NetworkSettings/WifiConnectionDetails.tsx:74]: Next week I will have a meeting with Brian and Shlok to improve this Wi-Fi setting's implementation.

**Review by ewagoner — APPROVED** [review]
